packages/smooth_app/lib/pages/preferences/user_preferences_settings.dart

```diff
@@ -54,9 +59,115 @@ class UserPreferencesSettings extends AbstractUserPreferences {
+    try {
+      await StoreRedirect.redirect(
+        androidAppId: 'org.openfoodfacts.scanner',
+        iOSAppId: 'org.openfoodfacts.scanner',
+      );
+    } on PlatformException {
+      final AppLocalizations appLocalizations = AppLocalizations.of(context);
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          content: Text(
+            '${appLocalizations.could_not_find} ${Platform.isIOS ? "app store" : "play store"}',
+            textAlign: TextAlign.center,
+          ),
+          behavior: SnackBarBehavior.floating,
+        ),
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    return Column(
+      children: <Widget>[
+        ListTile(
+          leading: Padding(
+            padding: const EdgeInsets.all(SMALL_SPACE),
+            child: SizedBox(
+              height: DEFAULT_ICON_SIZE,
+              width: DEFAULT_ICON_SIZE,
+              child: Image.asset(Platform.isIOS
+                  ? 'assets/app/app-store.png'
+                  : 'assets/app/playstore.png'),
+            ),
+          ),
+          title: Text(
+            appLocalizations.app_rating_dialog_positive_action,
+            style: const TextStyle(fontWeight: FontWeight.bold),
+          ),
+          onTap: () => _redirect(context),
+        ),
+        const SizedBox(
+          height: SMALL_SPACE,
+        ),
+        const UserPreferencesListItemDivider(),
+      ],
+    );
+  }
+}
+
+class _ShareWithFriends extends StatelessWidget {
+  const _ShareWithFriends();
+
+  Future<void> _shareApp(BuildContext context) async {
+    const String packageName = 'org.openfoodfacts.scanner';
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    // We need to provide a sharePositionOrigin to make the plugin work on ipad
+    final RenderBox? box = context.findRenderObject() as RenderBox?;
+    final String url = Platform.isIOS
+        ? 'market://details?id=$packageName'
+        : 'https://play.google.com/store/apps/details?id=$packageName';
+    try {
+      Share.share(
+        "${appLocalizations.discover_best_food_scanner}: 'Oen Food Facts - Food scanner' $url",
+        subject:
+            "${appLocalizations.discover_best_food_scanner}: 'Oen Food Facts - Food scanner'",
+        sharePositionOrigin: box!.localToGlobal(Offset.zero) & box.size,
+      );
+    } on PlatformException {
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          content: Text(
+            appLocalizations.error,
+            textAlign: TextAlign.center,
+          ),
+          behavior: SnackBarBehavior.floating,
+        ),
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: <Widget>[
+        ListTile(
+          leading: const Icon(Icons.share),
+          title: const Text(
+            'Share Open Food Facts',
```

Comment:
string to extract

Reply:
Should I remove it, or should we add it here also ! There was a picture in the issue #2551 ,.i saw that and did this 😄.

Reply:
I think @teolemon means with extract to put it into app_en.arb